### ISSUE TITLE
Q: Composer dynamic variable based on SetEnv

### ISSUE THREAD

**Issue by ogrosko** [composer/composer#5309]
Hello
I'm using different enviroment variables for my project. Set in vhost config as follow:

`SetEnv TYPO3_CONTEXT Development` -  for dev server
`SetEnv TYPO3_CONTEXT Testing` -  for test server
`SetEnv TYPO3_CONTEXT Production` -  for live server

In composer I set up post-update-cmd which runs cli command
`"scripts": {
		"post-update-cmd": [
			"@clearCache"
		],
		"clearCache": [
			"TYPO3_CONTEXT=[actuall context] ./typo3cms cache:flush"
		]
	}`

Is it possible to set [actuall context] in `"TYPO3_CONTEXT=[actuall context] ./typo3cms cache:flush"` dynamicly?
In other words: When I run `composer update` 
on dev server clearCache script should looks like `TYPO3_CONTEXT=Development ./typo3cms cache:flush`
on test server clearCache script should looks like `TYPO3_CONTEXT=Testing ./typo3cms cache:flush`

etc.

Thank you for advice

**Comment by staabm** [comment]
@ogrosko you could define this ENV e.g. in the shell of your server (in case you have 1 server per ENV)

**Comment by alcohol** [comment]
I would suggest looking at [phpdotenv](/vlucas/phpdotenv).

**Comment by ogrosko** [comment]
Thanks for advice
But how can I change it dynamically in composer.json?

**Comment by stof** [comment]
you don't need to change it dynamically in the composer.json. Just put nothing aboout it in your clear cache command, as composer forwards the current environment variables to the subporesses it launches

**Comment by ogrosko** [comment]
I have no access to subprocesses (e.g. ./typo3cms is vendor extension)
Isn't possible to do something like this?

`"someComposerVar": "somePHPScriptReturningSomeValue"`

and than use this variable in my script command
`TYPO3_CONTEXT={$someComposerVar} ./typo3cms help`

or am I absolutelly out ;)

**Comment by alcohol** [comment]
I think you do not understand how ENV variables are inherited.

```
TYPO3_CONTEXT=X composer update
```

The above would make this ENV available to the `typo3cms` script that runs in the `post-update-cmd` step.

**Comment by ogrosko** [comment]
ahaaaa great thank you @alcohol
